### PR TITLE
fix(backend): Allow empty namespace and remove default namespace. Fixes #8945. Fixes #8897. Fixes #8854.

### LINKED ISSUE #8854: [backend] Permission issues with api-server:2.0.0-beta.0
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
Installed following kubeflow manifests installation - master version (currently identical to tag v1.7.0-rc.0) of manifests with beta.0 instead of alpha.6 version of kubeflow pipeline components.
*  KFP version: 
2.0.0-beta.0
*  KFP SDK version: 
2.0.0b12

### Steps to reproduce

Run pipeline like described in [pipelines documentation](https://www.kubeflow.org/docs/components/pipelines/v2/author-a-pipeline/components/).
The call to `create_run_from_pipeline_func` fails with following error:
```
HTTP response body: {"error":"Failed to create a new v1beta1 run: Failed to create a run due to authorization error. Check if you have write permissions to namespace kubeflow-user-example-com: PermissionDenied: User 'user@example.com' is not authorized with reason:  (request: \u0026ResourceAttributes{Namespace:kubeflow-user-example-com,Verb:create,Group:,Version:,Resource:pipelines,Subresource:,Name:,}): Unauthorized access","code":7,"message":"Failed to create a new v1beta1 run: Failed to create a run due to authorization error. Check if you have write permissions to namespace kubeflow-user-example-com: PermissionDenied: User 'user@example.com' is not authorized with reason:  (request: \u0026ResourceAttributes{Namespace:kubeflow-user-example-com,Verb:create,Group:,Version:,Resource:pipelines,Subresource:,Name:,}): Unauthorized access","details":[{"@type":"type.googleapis.com/google.rpc.Status","code":7,"message":"User 'user@example.com' is not authorized with reason:  (request: \u0026ResourceAttributes{Namespace:kubeflow-user-example-com,Verb:create,Group:,Version:,Resource:pipelines,Subresource:,Name:,})"}]}
```
The roles and binding created from manifest look correct. After some digging I found that problem comes from [here](https://github.com/kubeflow/pipelines/blob/2.0.0-beta.0/backend/src/apiserver/server/run_server.go#L132) :
```go
	resourceAttributes := &authorizationv1.ResourceAttributes{
		Namespace: run.Namespace,
		Verb:      common.RbacResourceVerbCreate,
		Resource:  common.RbacResourceTypePipelines,
	}
	if err := s.resourceManager.IsAuthorized(ctx, resourceAttributes); err != nil {
		return nil, util.Wrapf(err, "Failed to create a run due to authorization error. Check if you have write permissions to namespace %s", run.Namespace)
	}
```
The `resourceAttributes` doesn't specify `Group`.
After adding `Group:     common.RbacPipelinesGroup,` (`common.RbacPipelinesGroup` being `pipelines.kubeflow.org`) to attributes, building the docker image and using it instead of `gcr.io/ml-pipeline/api-server:2.0.0-beta.0` I run to exactly same problem [in other place](https://github.com/kubeflow/pipelines/blob/2.0.0-beta.0/backend/src/apiserver/server/pipeline_server.go#L122). Again, the group added, image built, pushed, used, the pipeline is running.
Problem could in other places as well.

### Expected result

Pipeline runs

---

Impacted by this bug? Give it a 👍.

### LINKED ISSUE #8897: [backend] pipelines not accessible after upgrade to 2.0.0-beta.0
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
Full Kubeflow deployment managed by K8S team.
*  KFP version: kubeflow 1.6 and kubeflow pipelines 2.0.0-beta.0
*  KFP SDK version: 

### Steps to reproduce
This issue started occurring after pipelines upgrade to 2.0.0-beta.0 version from 2.0.0b6.  
This is on a multi user setup.   After this upgrade, I cannot list pipelines.   But I can access runs, experiments etc with no issues.

`I0225 00:06:06.131318       7 interceptor.go:29] /api.PipelineService/ListPipelinesV1 handler starting
I0225 00:06:06.131592       7 resource_manager.go:1781] Getting user identity
I0225 00:06:06.131618       7 resource_manager.go:1802] User: <****masked***>, ResourceAttributes: &ResourceAttributes{Namespace:,Verb:get,Group:pipelines.kubeflow.org,Version:v1beta1,Resource:pipelines,Subresource:,Name:,}
I0225 00:06:06.131645       7 resource_manager.go:1803] Authorizing request
I0225 00:06:06.136920       7 resource_manager.go:1832] PermissionDenied: User '***Masked*****' is not authorized with reason:  (request: &ResourceAttributes{Namespace:,Verb:get,Group:pipelines.kubeflow.org,Version:v1beta1,Resource:pipelines,Subresource:,Name:,}): Unauthorized access
E0225 00:06:06.136950       7 error.go:272] Unauthorized access
github.com/kubeflow/pipelines/backend/src/apiserver/resource.(*ResourceManager).IsAuthorized
	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/resource/resource_manager.go:1826
github.com/kubeflow/pipelines/backend/src/apiserver/server.(*PipelineServer).checkAccess
	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/server/pipeline_server.go:1018
github.com/kubeflow/pipelines/backend/src/apiserver/server.(*PipelineServer).listPipelines
	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/server/pipeline_server.go:370
github.com/kubeflow/pipelines/backend/src/apiserver/server.(*PipelineServer).ListPipelinesV1
	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/server/pipeline_server.go:430
github.com/kubeflow/pipelines/backend/api/v1beta1/go_client._PipelineService_ListPipelinesV1_Handler.func1
	/go/src/github.com/kubeflow/pipelines/backend/api/v1beta1/go_client/pipeline.pb.go:2108
main.apiServerInterceptor
	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/interceptor.go:30
github.com/kubeflow/pipelines/backend/api/v1beta1/go_client._PipelineService_ListPipelinesV1_Handler
	/go/src/github.com/kubeflow/pipelines/backend/api/v1beta1/go_client/pipeline.pb.go:2110
`

URL looks like this https://***masked***/_/pipeline/?ns=mynamespace#/pipelines
For list pipelines looks like Namespace shows empty in the resource attributes although it's available in the url.
I0224 23:46:50.574897       7 resource_manager.go:1802] User: ***masked***, ResourceAttributes: &ResourceAttributes{Namespace:,Verb:get,Group:pipelines.kubeflow.org,Version:v1beta1,Resource:pipelines,Subresource:,Name:,}

Where as If I access runs https://***masked***/_/pipeline/?ns=mynamespace#/runs , this works fine. 
I0225 00:01:27.712083       7 resource_manager.go:1802] User: ***masked***, ResourceAttributes: &ResourceAttributes{Namespace:mynamespace,Verb:list,Group:pipelines.kubeflow.org,Version:v1beta1,Resource:runs,Subresource:,Name:,}

I see that https://github.com/kubeflow/pipelines/blame/master/backend/src/apiserver/server/pipeline_server.go#L362 has undergone changes with beta release.  

### Expected result

User should be able to access pipelines page with no issues. 

### Materials and Reference

---

Impacted by this bug? Give it a 👍.

### LINKED ISSUE #8945: [backend] defaultNamespace cmd argument for api-server since beta.0
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
From manifests, using version beta.0 images instead of alpha.6
*  KFP version: 
2.0.0-beta.0
*  KFP SDK version: 
2.0.0b12 (but it is irrelevant I think for this)

### Steps to reproduce

Apply `ml-pipeline` deployment on a cluster where there is no default experiment yet created in mysql. The pod will fail to start with the error:
```
main.go:76] Failed to create default experiment. Err: Failed to create the default experiment: Invalid input error: Namespace cannot be empty
```

### Expected result

Pod starting creating the default experiment.

### Materials and Reference

In pull request #8633 there was new command line argument for api-server added named `defaultNamespace` with an empty string as default value. This value is used for creation of new default experiment if it is not yet there - you will not have this issue if you are just updating because the default experiment will be already there (in mysql). There is an empty string check [here](https://github.com/kubeflow/pipelines/blob/2.0.0-beta.0/backend/src/apiserver/resource/resource_manager.go#L1027) throwing the error.

I guess as a workaround I can manually insert an experiment and default experiment id into the mysql with some reasonable values and pod will start (still to try).

I assume `Dockerfile`'s `CMD` has to be adjusted to provide this argument when executing api-server - not sure what the value should be though - what is the indented value?

---

Impacted by this bug? Give it a 👍.

### PR DESCRIPTION
**Pull request by gkcalat**
**Related issues:**
- #8854
- #8897
- #8945

**Changes:**

- Set empty string instead of `defaultNamesppace`
- All `GET` calls with missing or empty namespace and experimentId should return all resources (single-user) and fail (multi-user)
- All `POST` calls with missing or empty namespace and experimentId should create resources in the default experiment (single-user) or fail (multi-user)

- Refactored:
   - `CreateRun` and `CreateJob` verify input values and authentication context at the `server` layer, while `resourceManager` only manipulates resources.
   - Changed unit tests for `CreateRun` into a tabular format for better readability and easier maintenance
   - Moved a unit test for creating resources without an experiment to the server package

**Tested:**
- [x] Standalone @ GKE 1.25
- [x] Full Kubeflow v1.7.0 @ GKE 1.25

/hold

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by gkcalat** [comment]
retest

File: backend/src/apiserver/model/default_experiment.go [hunk 1]
```diff
@@ -15,5 +15,6 @@
 package model
 
 type DefaultExperiment struct {
-	DefaultExperimentId string `gorm:"column:DefaultExperimentId; not null; primary_key;"`
+	DefaultExperimentId string `gorm:"column:DefaultExperimentId; not null;"`
+	Namespace           string `gorm:"column:Namespace; not null; primary_key;"`
```
[the remark this answers is not in the document]
Comment (gkcalat) [reply backend/src/apiserver/model/default_experiment.go]: You are right. `git push` failed yesterday. I reverted the change.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**